compute_tools/src/bin/compute_ctl.rs

```diff
@@ -284,6 +332,22 @@ fn main() -> Result<()> {
         exit_code = ecode.code()
     }
 
+    // Terminate the vm_monitor so it releases the file watcher on
+    // /sys/fs/cgroup/neon-postgres.
+    // Note: the vm-monitor only runs on linux because it requires cgroups.
+    cfg_if::cfg_if! {
+        if #[cfg(target_os = "linux")] {
+            if let Some(handle) = vm_monitor {
+                token.cancel();
+                handle.abort();
```

Comment:
This looks weird, would rather have one OR the other, not both.

Assuming you have *any* code running behind that handle awaiting to get a wakeup from `token.cancelled()` it will be "delievered asynchronously", which means that by the time it "get's" there, you've already aborted the handle.

But perhaps you are distributing the same token to all spawned tasks and then this would make sense. I'd perhaps handle with a CancellationToken within the first task you spawn, because then all spawned tasks can use the same one, and there does not need to be one in this "whole application scope".

Reply:
Yeah, the token is being handed to every task spawned by the monitor. So the first statement cancels all the spawned tasks. Then, the the second statement cancels the actual task spawned in `compute_ctl` itself. I can add some comments.

Reply:
Clarified this logic in 88e6dc023a472cd75b7c0e749d77384d8827b099

Reply:
Ok, resolving.